src/settingsholder.cpp

```diff
@@ -186,16 +191,19 @@ bool SettingsHolder::hasAddonSetting(const AddonSettingQuery& query) {
 }
 
 QString SettingsHolder::getAddonSetting(const AddonSettingQuery& query) {
-  if (!hasAddonSetting(query)) return query.m_defaultValue;
-
   QString key = getAddonSettingKey(query);
+
+  if (!m_settings.contains(key)) return query.m_defaultValue;
```

Comment:
We can delete `hasAddonSetting`, because with this change it's not being used anymore.